baybe/utils/interval.py

```diff
@@ -41,40 +45,66 @@ class Interval:
     """The upper end of the interval."""
 
     @upper.validator
-    def _validate_order(self, _: Any, value: float):  # noqa: DOC101, DOC103
+    def _validate_order(self, _: Any, upper: float) -> None:  # noqa: DOC101, DOC103
         """Validate the order of the interval bounds.
 
         Raises:
             ValueError: If the upper end is not larger than the lower end.
         """
-        if value <= self.lower:
+        if upper <= self.lower:
             raise ValueError(
-                f"The upper interval bound (provided value: {value}) must be larger "
+                f"The upper interval bound (provided value: {upper}) must be larger "
                 f"than the lower bound (provided value: {self.lower})."
             )
 
     @property
-    def is_finite(self):
+    def is_closed(self) -> bool:
```

Comment:
This naming could yield confusion: Note that a closed interval is typically interpreted as an interval containing both end points, while an open interval is one that does NOT contain either of the endpoints. For example, the interval (0,1) not containing the numbers 0 and 1 is an open but finite interval. In this terminology, this would however be a *closed* interval which is not the correct term here. I guess what you mean here is better called bounded?

Reply:
Unfortunately, we seem to have used this term previously, so I guess this might yield problems resp. force us to already get rid of the backwards compatibility?

Reply:
OMG, thanks for pointing out!! As a true German would say: _"Before all the trees, I did not see the wood anymore! ... oder so ähnlich"_ 😄 Was so focused on replacing the previous ambiguous names that I did not pay attention here and really forgot what an open interval is 😭  So my clear suggestion would be: let's use the precise mathematical naming for our properties, no matter if this now requires another refactoring. I can take care of preparing the draft and will ping you once done. Does this sound right?

Reply:
Works for me 😄

Reply:
So, how about now? Sticking purely to the mathematical definitions of boundedness, hopefully =)

Reply:
Gucci now :)